src/coreComponents/physicsSolvers/fluidFlow/RegionMultiphaseStatistics.cpp

```diff
@@ -0,0 +1,906 @@
+    regData[Statistics::MINIMUM_TEMPERATURE] = LvArray::math::min( regData[Statistics::MINIMUM_TEMPERATURE], minData[Statistics::MINIMUM_TEMPERATURE] );
+    regData[Statistics::MAXIMUM_TEMPERATURE] = LvArray::math::max( regData[Statistics::MAXIMUM_TEMPERATURE], -minData[Statistics::MAXIMUM_TEMPERATURE] );
+
+    regData[Statistics::AVERAGE_PRESSURE] += sumData[Statistics::AVERAGE_PRESSURE];
+    regData[Statistics::AVERAGE_TEMPERATURE] += sumData[Statistics::AVERAGE_TEMPERATURE];
+    for( integer propIndex = Statistics::PHASE_PORE_VOLUME; propIndex < Statistics::END; ++propIndex )
+    {
+      regData[propIndex] += sumData[propIndex];
+    }
+  }
+};
+
+RegionMultiphaseStatistics::RegionMultiphaseStatistics( const string & name,
+                                                        Group * const parent ):
+  Base( name, parent )
+{
+  registerWrapper( viewKeyStruct::regionNamesString(), &m_regionNames ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "The list of names of the regions" );
+
+  registerWrapper( viewKeyStruct::regionIdentifiersString(), &m_regionIdentifiers ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "The list of integer identifiers of the regions" );
+
+  registerWrapper( viewKeyStruct::propertyNamesString(), &m_propertyNames ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "The list of names of properties to report" );
+}
+
+void RegionMultiphaseStatistics::postProcessInput()
+{
+  Base::postProcessInput();
+
+  // Make sure we have the same number of region names and region indices
+  GEOS_ERROR_IF_NE_MSG( m_regionNames.size(), m_regionIdentifiers.size(),
+                        GEOS_FMT( "The number of region names ({}) is different from the number of region identifiers ({}).",
+                                  m_regionNames.size(), m_regionIdentifiers.size() ) );
+
+  // Convert the property names to integers
+  m_propertyNameTypes.clear();
+  localIndex const propSize = m_propertyNames.size();
+  if( propSize == 0 )
+  {
+    m_propertyNameTypes.emplace_back( PropertyNameType::Pressure );
+    m_propertyNameTypes.emplace_back( PropertyNameType::Temperature );
+    m_propertyNameTypes.emplace_back( PropertyNameType::StaticPoreVolume );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PoreVolume );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhaseVolumeFraction );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhasePoreVolume );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhaseMass );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhaseDensity );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhaseViscosity );
+    m_propertyNameTypes.emplace_back( PropertyNameType::PhaseComponentMass );
+    m_propertyNameTypes.emplace_back( PropertyNameType::ComponentMass );
```

Comment:
In order to automatically iterate over enum labels :
```suggestion
    constexpr auto labelCount = EnumStrings< PropertyNameType >::count;
    for( integer i = 0; i < labelCount; ++i )
    {
      m_propertyNameTypes.emplace_back( i );
    }
```

... we could add a `getEnumStringsCount( ENUM const )` in the `ENUM_STRINGS()` macro :
```
  inline constexpr size_t getEnumStringsCount( ENUM const )           \
  {                                                                   \
    constexpr char const * arr[] { __VA_ARGS__ };                     \
    constexpr auto s = std::end( arr ) - std::begin( arr );           \
    return s;                                                         \
  }                                                                   \
```

... and a constexpr `count` in `EnumStrings` :
```
  inline static constexpr size_t count = getEnumStringsCount( enum_type{} );
```